metrics/lib/json.bash

```diff
@@ -129,6 +131,43 @@ metrics_json_end_of_system() {
 	system_index=$(( ${#json_result_array[@]}))
 }
 
+# Add a array of number of physical sockets and hypervisor per node within the Kubernetes cluster
+metrics_json_num_sockets_node(){
+
+	metrics_json_start_nested_array
+
+	# grab pods in the stats daemonset
+	while read -u 3 name node; do
+		# gather node information about number of sockets and its hypervisor if any
+		local num_sockets=$(kubectl exec -ti $name -- sh -c "cat /proc/cpuinfo | grep 'physical id' | sort -u | wc -l" | sed 's/\r//')
+		local node_hypervisor=false
+		kubectl exec -ti $name -- sh -c "grep -q ^flags.*\ hypervisor\  /proc/cpuinfo"
+
+		if [ $? -eq 0 ]; then
+			node_hypervisor=true
+		fi
+
+		info "Num socket at current node: $num_sockets"
+		info "Hypervisor at current node: $node_hypervisor"
+
+		local util_json="$(cat << EOF
+		{
+			"node": "${node}",
+			"num_sockets": ${num_sockets},
+			"hypervisor" : "${node_hypervisor}"
+		}
+EOF
+		)"
+
+
+		metrics_json_add_nested_array_element "$util_json"
+
+	done 3< <(kubectl get pods --selector name=stats-pods -o json | jq -r '.items[] | "\(.metadata.name) \(.spec.nodeName)"')
+
+	# write node's information into json output
+	metrics_json_end_nested_array "sockets_per_node"
```

Comment:
and to be consistent, let's change "sockets_per_node" to "socketsPerNode"

Reply:
Done.

Although, talking about consistency there are many keys that don't follow this naming, but this can be addressed in a separate PR.